_posts/2019-11-27-update-sitemaps-during-heroku-deploy.markdown

```diff
@@ -0,0 +1,122 @@
+
+Since the blog it's a gem, we also need to make sure that whatever tool we use to generate the sitemap would also cover new blog posts.
+
+In this article I'll show you my journey to figure out how to made everything work together.
+
+<!--more-->
+
+## First of all: what is a sitemap?
+
+According to [Google](https://support.google.com/webmasters/answer/156184?hl=en) itself:
+> "A sitemap is a file where you provide information about the pages, videos, and other files on your site, and the relationships between them. Search engines like Google read this file to more intelligently crawl your site."
+
+It tells to search engines which pages and files you think are important in your site, and also provides valuable information about these files: for example, when the page was last updated, how often the page is changed, and others.
+
+So it's important to keep it always up-to-date.
+
+## Configuring the Gem
+
+Listing all routes of a website and add them to a XML file seems to be a repetitive task, so I thought that we might have a gem for it already. And we do. It's the [`sitemap_generator`](https://github.com/kjvarga/sitemap_generator) gem, which does exactly what we need: provide a few rails tasks to generate and refresh the application sitemap.
+
+Installing the gem is pretty easy, just add it to your `Gemfile`:
+
+`gem 'sitemap_generator'`
+
+run the `bundle install` and then run this task:
+
+`rake sitemap:install`
+
+that will create a config file with some information to be used by the gem, like the sample below:
+
+```
+require 'sitemap_generator'
+
+SitemapGenerator::Sitemap.default_host = 'http://example.com'
+SitemapGenerator::Sitemap.create do
+  add '/home', :changefreq => 'daily', :priority => 0.9
+  add '/contact_us', :changefreq => 'weekly'
+end
+```
+
+As you can see, inside the `Sitemap.create` block we need to add all routes of our site.
+
+If your site has only static content, your job here it's done. You just need to use the rails task from the sitemap_generator itself `rake sitemap:refresh` and a `sitemap.xml` file will be added to the `public` folder.
+
+But in my case new routes will be added every time that someone writes a new article. And as we know, the blog it's a Jekyll application,
+so we can't get the routes using the rake task provided by the `sitemap_generator` gem.
+
+## Generating the sitemap index
+
+To solve this issue, my idea was to change the `sitemap_generator` config to instead of generate just one `sitemap.xml` file, it would create two sitemaps (one for the blog and one for the website) and add them to a index file.
```

Comment:
```suggestion
To solve this issue, my idea was to change the `sitemap_generator` config to create two sitemaps (one for the blog and one for the website) and add them to an index file instead of generating just one `sitemap.xml` file.
```